tests/xtensor/test_shape.py

```diff
@@ -369,3 +371,213 @@ def test_squeeze_errors():
+    x = xtensor("x", dims=(), shape=())
+    y = expand_dims(expand_dims(x, "a"), "b")
+    fn = xr_function([x], y)
+    expected = xr_arange_like(x).expand_dims("a").expand_dims("b")
+    xr_assert_allclose(fn(xr_arange_like(x)), expected)
+
+
+def test_expand_dims_symbolic_size():
+    """Test expand_dims with symbolic sizes."""
+
+    # Symbolic size=1: same as default
+    size_sym_1 = scalar("size_sym_1", dtype="int64")
+    x = xtensor("x", dims=("a", "b"), shape=(2, 3))
+    y = expand_dims(x, "batch", size=size_sym_1)
+    fn = xr_function([x, size_sym_1], y, on_unused_input="ignore")
+    x_test = xr_arange_like(x)
+    xr_assert_allclose(fn(x_test, 1), x_test.expand_dims("batch"))
+
+    # Test using symbolic size from an existing dimension of the same tensor
+    # This verifies that expand_dims can use the size of one dimension to create another
+    x = xtensor(dims=("a", "b", "c"))
+    y = expand_dims(x, "d", size=x.sizes["b"])
+    fn = xr_function([x], y)
+    x_test = xr_arange_like(xtensor(dims=x.dims, shape=(2, 3, 5)))
+    res = fn(x_test)
+    expected = x_test.expand_dims({"d": 3})  # 3 is the size of dimension "b"
+    xr_assert_allclose(res, expected)
+
+    # Test broadcasting with symbolic size from a different tensor
+    x = xtensor("x", dims=("a", "b"), shape=(2, 3))
+    other = xtensor("other", dims=("c",), shape=(4,))
+    y = expand_dims(x, "batch", size=other.sizes["c"])
+    fn = xr_function([x, other], y)
+    x_test = xr_arange_like(x)
+    other_test = xr_arange_like(other)
+    res = fn(x_test, other_test)
+    expected = x_test.expand_dims(
+        {"batch": 4}
+    )  # 4 is the size of dimension "c" in other
+    xr_assert_allclose(res, expected)
+
+    # Test behavior with symbolic size > 1
+    # NOTE: This test documents our current behavior where expand_dims broadcasts to the requested size.
+    # This differs from xarray's behavior where expand_dims always adds a size-1 dimension.
+    size_sym_4 = scalar("size_sym_4", dtype="int64")
+    x = xtensor("x", dims=("a", "b"), shape=(2, 3))
+    y = expand_dims(x, "batch", size=size_sym_4)
+    fn = xr_function([x, size_sym_4], y, on_unused_input="ignore")
+    x_test = xr_arange_like(x)
+    res = fn(x_test, 4)
+    # Our current behavior: broadcasts to size 4
+    expected = x_test.expand_dims({"batch": 4})
+    xr_assert_allclose(res, expected)
+    # xarray's behavior would be:
+    # expected = x_test.expand_dims("batch")  # always size 1
+    # xr_assert_allclose(res, expected)
+
+    # Test using symbolic size from a reduction operation
+    x = xtensor("x", dims=("a", "b"), shape=(2, 3))
+    reduced = x.sum("a")  # shape: (b: 3)
+    y = expand_dims(x, "batch", size=reduced.sizes["b"])
+    fn = xr_function([x], y)
+    x_test = xr_arange_like(x)
+    res = fn(x_test)
+    expected = x_test.expand_dims({"batch": 3})  # 3 is the size of dimension "b"
+    xr_assert_allclose(res, expected)
+
+    # Test chaining expand_dims with symbolic sizes
```

Comment:
These tests are trying really hard to not just use a symbolic size variable, why?